Mathlib/Data/Matrix/Vec.lean

```diff
@@ -0,0 +1,145 @@
+/-
+Copyright (c) 2025 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.LinearAlgebra.Matrix.Trace
+import Mathlib.Data.Matrix.Kronecker
+import Mathlib.Data.Matrix.Hadamard
+
+/-! # Vectorization of matrices
+
+This file defines `Matrix.vec A`, the vectorization of a matrix `A`,
+formed by stacking the columns of A into a single large column vector.
+
+Since mathlib indices matrices by arbitrary types rather than `Fin n`,
+the result of `Matrix.vec` on `A : Matrix m n R` is indexed by `n × m`.
+The `Fin (n * m)` interpretation can be restored by composing with `finProdFinEquiv.symm`:
+```lean
+-- ![1, 2, 3, 4]
+#eval vec !![1, 3; 2, 4] ∘ finProdFinEquiv.symm
+```
+
+While it may seem more natural to index by `m × n`, keeping the indices in the same order,
+this would amount to stacking the rows into one long row, and goes against the literature.
+If you want this function, you can write `Matrix.vec Aᵀ` instead.
+
+### References
+
+* [Wikipedia](https://en.wikipedia.org/wiki/Vectorization_(mathematics))
+-/
+namespace Matrix
+
+variable {ι l m n p R S}
+
+/-- All the matrix entries, arranged into one column. -/
+def vec (A : Matrix m n R) : n × m → R :=
+  fun ij => A ij.2 ij.1
+
+theorem vec_transpose (A : Matrix m n R) : vec Aᵀ = vec A ∘ Prod.swap := rfl
+
+theorem vec_eq_uncurry (A : Matrix m n R) : vec A = Function.uncurry fun i j => A j i := rfl
```

Comment:
perhaps a version mentioning `flip` too? optional

Reply:
I think the `of` version now covers this?